metrics_metadata.go

```diff
@@ -161,6 +161,34 @@ func (c *Client) GetMetric(ctx context.Context, name string) (*metrics_metadata.
 	return finalMetric, err
 }
 
+// UpdateMetric creates or updates a metric
+func (c *Client) CreateUpdateMetric(ctx context.Context, name string, cumr *metrics_metadata.CreateUpdateMetricRequest) (*metrics_metadata.Metric, error) {
+	payload, err := json.Marshal(cumr)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.doRequest(ctx, "PUT", MetricAPIURL+"/"+name, nil, bytes.NewReader(payload))
```

Comment:
Is the MetricAPIURL guaranteed to be normalized to have to trailing `/`?  If not, best to use stdlib url path join functions here instead of manual concatenation.